File: Sheets/src/neetcode/practice/Bit_Manipulation/_1_singleNumber.java
package neetcode.practice.Bit_Manipulation;

public class _1_singleNumber {
  public static int singleNumber(int[] nums) {
    int ans = nums[0], n = nums.length;
    for (int i = 1; i < n; i++) ans ^= nums[i];
    return ans;
  }
}

File: Sheets/src/neetcode/practice/Bit_Manipulation/_1_SingleNumber.cc
#include <vector>

class Solution {
public:
  int singleNumber(std::vector<int> &nums) {
    std::size_t n = nums.size();
    if (n == 1)
      return nums[0];

    int unique = nums[0];
    for (std::size_t i = 1; i < n; i++)
      unique ^= nums[i];

    return unique;
  }
};
